core/web/api/links.py

```diff
@@ -66,3 +66,51 @@ def multiupdate(self):
             updated.append(link.id)
 
         return render({"updated": updated})
+
+    @route("/", methods=['POST'])
+    @requires_permissions('write')
+    def new(self):
+        """Create a new link
+
+        Create a new link from the JSON object passed in the ``POST`` data.
+
+        :<json object params: JSON object containing object ids to link
+        """
+        params = request.json
+        type_src = params.pop('type_src', None)
+        type_dst = params.pop('type_dst', None)
+        db_objects_src = None
+        db_objects_dst = None
+
+        if type_src == 'observable':
+            db_objects_src = observables.Observable
+        elif type_src == 'entity':
+            db_objects_src = entities.Entity
+        elif type_src == 'indicator':
+            db_objects_src = indicators.Indicator
+
+        if type_dst == 'observable':
+            db_objects_dst = observables.Observable
+        elif type_dst == 'entity':
+            db_objects_dst = entities.Entity
+        elif type_dst == 'indicator':
+            db_objects_dst = indicators.Indicator
+
+        if db_objects_src and db_objects_dst:
+            src = db_objects_src.objects.get(id=params.pop("link_src"))
```

Comment:
This will break if "link_src" is not specified in the request. Please check that all mandatory parameters are present at the start of the function and return HTTP errors if not. 

You may also want to return a 404 if `src` or `dst` is not found.

Also, it seems that you want to use `get` instead of `pop` unless there's a strong reason to remove the `'link_src'` key from the `params` dict. This is valid for other places where you use `pop` too.